build/builder.go

```diff
@@ -998,5 +1005,14 @@ func (e deltaBranchSetError) Error() string {
 	return fmt.Sprintf("repository metadata in shard %q contains a different set of branch names than what was requested, which is unsupported in a delta shard build. old: %+v, new: %+v", e.shardName, e.old, e.new)
 }
 
+type deltaIndexOptionsMismatchError struct {
+	shardName  string
+	newOptions Options
+}
+
+func (e deltaIndexOptionsMismatchError) Error() string {
```

Comment:
nit: normally easier to always make Error on a pointer receiver. I see above we didn't do this, but when switching on error values people normally use pointers consistently since it can lead to a go footgun with errors.Is/As

Reply:
updated